src/Elliptic/Systems/Poisson/Observe.hpp

```diff
@@ -0,0 +1,124 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "Domain/Tags.hpp"
+#include "Elliptic/Systems/Poisson/Tags.hpp"
+#include "IO/Observer/Actions.hpp"
+#include "IO/Observer/ArrayComponentId.hpp"
+#include "IO/Observer/ObservationId.hpp"
+#include "IO/Observer/ObserverComponent.hpp"
+#include "IO/Observer/ReductionActions.hpp"
+#include "IO/Observer/VolumeActions.hpp"
+#include "NumericalAlgorithms/LinearOperators/DefiniteIntegral.hpp"
+#include "NumericalAlgorithms/LinearSolver/Tags.hpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Tags.hpp"
+#include "Utilities/Functional.hpp"
+#include "Utilities/MakeString.hpp"
+#include "Utilities/Numeric.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace Poisson {
+namespace Actions {
+
+/*!
+ * \brief Temporary action to observe volume and reduction data for the Poisson
+ * system
+ */
+struct Observe {
+  template <typename... DbTags, typename... InboxTags, typename Metavariables,
+            size_t Dim, typename ActionList, typename ParallelComponent>
+  static auto apply(db::DataBox<tmpl::list<DbTags...>>& box,
+                    tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+                    Parallel::ConstGlobalCache<Metavariables>& cache,
+                    const ElementIndex<Dim>& array_index,
+                    const ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/) noexcept {
+    const auto& iteration_id = get<LinearSolver::Tags::IterationId>(box);
+    const auto& mesh = get<Tags::Mesh<Dim>>(box);
+    const std::string element_name = MakeString{} << ElementId<Dim>(array_index)
+                                                  << '/';
+
+    // Retrieve the current numeric solution
+    const auto& field = get<Poisson::Field>(box);
+
+    // Compute the analytic solution
+    const auto& inertial_coordinates =
+        db::get<::Tags::Coordinates<Dim, Frame::Inertial>>(box);
+    const auto field_analytic = get<Poisson::Field>(
+        Parallel::get<typename Metavariables::analytic_solution_tag>(cache)
+            .field_variables(inertial_coordinates));
+
+    // Compute error between numeric and analytic solutions
+    DataVector field_error = get(field) - get(field_analytic);
+
+    // Retrieve linear residual
+    // We have to reduce it here again, even though it should be the same on all
```

Comment:
This is probably because I am unfamiliar with the code, but what does "it" refer to in this comment? Also, is the residual itself linear or could "linear residual" be better phrased as "The residual from the linear solver"?

Reply:
rephrased it